datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -284,6 +333,34 @@ impl<T: FieldValues> Cursor for FieldCursor<T> {
         self.offset += 1;
         t
     }
+
+    fn slice(&self, offset: usize, length: usize) -> Self {
+        let FieldCursor {
+            values,
+            offset: _,
+            null_threshold,
+            options,
+        } = self;
+
+        let null_threshold = match self.options.nulls_first {
+            true => null_threshold.saturating_sub(offset),
+            false => {
+                let shorter_len = self.values.len().saturating_sub(offset + length + 1);
+                null_threshold.saturating_sub(offset.saturating_sub(shorter_len))
```

Comment:
Now that I think about this more, I am unsure why `null_threshold.saturating_sub(offset)` is incorrect